src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp

```diff
@@ -267,65 +223,59 @@ void TypeLookupManager::remove_remote_endpoints(
 }
 
-StatefulWriter* TypeLookupManager::get_builtin_request_writer()
+fastrtps::rtps::SampleIdentity TypeLookupManager::get_type_dependencies(
+        const fastrtps::types::TypeIdentifierSeq& id_seq) const
 {
-    return builtin_request_writer_;
-}
+    fastrtps::rtps::SampleIdentity id = INVALID_SAMPLE_IDENTITY;
 
-StatefulWriter* TypeLookupManager::get_builtin_reply_writer()
-{
-    return builtin_reply_writer_;
-}
+    TypeLookup_getTypeDependencies_In in;
+    //TODO with xtypes::TypeIdentifierSeq
+    // in.type_ids() = id_seq;
+    (void) id_seq;
+    TypeLookup_RequestPubSubType type;
+    TypeLookup_Request* request = static_cast<TypeLookup_Request*>(type.createData());
+    request->data().getTypeDependencies(in);
 
-WriterHistory* TypeLookupManager::get_builtin_request_writer_history()
-{
-    return builtin_request_writer_history_;
-}
-
-WriterHistory* TypeLookupManager::get_builtin_reply_writer_history()
-{
-    return builtin_reply_writer_history_;
+    if (send_request(*request))
+    {
+        id = get_rtps_sample_identity(request->header().requestId());
+    }
+    type.deleteData(request);
+    return id;
 }
 
-StatefulReader* TypeLookupManager::get_builtin_request_reader()
+fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
+        const fastrtps::types::TypeIdentifierSeq& id_seq) const
 {
-    return builtin_request_reader_;
-}
+    fastrtps::rtps::SampleIdentity id = INVALID_SAMPLE_IDENTITY;
 
-StatefulReader* TypeLookupManager::get_builtin_reply_reader()
-{
-    return builtin_reply_reader_;
-}
+    TypeLookup_getTypes_In in;
+    //TODO with xtypes::TypeIdentifierSeq
+    // in.type_ids() = id_seq;
+    (void) id_seq;
+    TypeLookup_RequestPubSubType type;
+    TypeLookup_Request* request = static_cast<TypeLookup_Request*>(type.createData());
+    request->data().getTypes(in);
 
-ReaderHistory* TypeLookupManager::get_builtin_request_reader_history()
-{
-    return builtin_request_reader_history_;
+    if (send_request(*request))
+    {
+        id = get_rtps_sample_identity(request->header().requestId());
+    }
+    type.deleteData(request);
+    return id;
 }
 
-ReaderHistory* TypeLookupManager::get_builtin_reply_reader_history()
+ReturnCode_t TypeLookupManager::async_get_type(
+        xtypes::TypeInformation,
+        GuidPrefix_t,
+        AsyncGetTypeCallback&)
 {
-    return builtin_reply_reader_history_;
+    return RETCODE_OK;
 }
 
-/* TODO Implement if security is needed.
- #if HAVE_SECURITY
-   bool TypeLookupManager::pairing_remote_reader_with_local_writer_after_security(
-        const GUID_t& local_writer,
-        const ReaderProxyData& remote_reader_data)
-   {
-
-   }
-
-   bool TypeLookupManager::pairing_remote_writer_with_local_reader_after_security(
-        const GUID_t& local_reader,
-        const WriterProxyData& remote_writer_data)
-   {
-
-   }
- #endif
- */
 bool TypeLookupManager::create_endpoints()
 {
+    EPROSIMA_LOG_INFO(TYPELOOKUP_SERVICE, "Initializing TypeLookup Service");
```

Comment:
I do not think this log should be kept. Before, only when a specific configuration was set, the TypeLookup service was initialized so this log would be helpful to check that it was initialized if the corresponding QoS was set. As the service is going to be initialized by default, I do not think this log provides meaningful information.

Reply:
Those logs provide more meaningful information because they express the correct initialization of a builtin endpoint. They could be useful for debug purposes. The removed log, was just providing info without undertaking any step that could fail previously.